### PR TITLE
Remove Tabix

### LINKED ISSUE #210: Removing wormtable dependency
This is a meta-issue collecting the steps we need to take to remove the wormtable dependency. Some of these may be turned into several separate issues, if they turn out to be sufficiently complex.

~~1. Merge #194 and #208, and start building data driven tests for the HtslibVariantSet. Once we are satisfied that it works well, remove the existing TabixVariantSet, and update the `variantSetFactory`.~~
~~2. Perform some performance analysis on the WormtableVariantSet and the HtslibVariantSet using the ga4gh-example-data. This now contains VCF versions of 1000g_2011 and 1000g_2013 variant sets, so we'll want to get some basic throughput stats on (a) wormtable; (b) plain VCF; and (c) BCF. If performance of the htslib backend is similar to or better than the wormtable backend, then we are free to continue with ripping out wormtable. ~~
~~3. Replace the Wormtable test fixture in tests/test_backends.py with other tests. Too much is currently tested in one place here. We should put the majority of the tests of variant sets (including data consistency and search semantics tests) into datadriven/test_variants.py. These should be fairly straight forward to convert. Paging and other issues handled exclusively by the backend module should be tested in unit/test_backends.py, perhaps using a simulated backend (see #202).~~
~~4. Replace other dependencies on the WormtableTestFixture in the testing infrastructure with a simulated backend.~~
5. Update documentation, example data, and all the rest to fully purge wormtable from the source.

### PR DESCRIPTION
**Pull request by dcolligan**
Does not close #210

### CODE REVIEW

File: ga4gh/datamodel/variants.py [hunk 1]
```diff
@@ -51,17 +51,13 @@ def convertVCFGenotype(vcfGenotype, vcfPhaseset):
 
 
 def variantSetFactory(variantSetId, relativePath):
-    if variantSetId.endswith(".wt"):
-        return WormtableVariantSet(variantSetId, relativePath)
-    else:
-        return TabixVariantSet(variantSetId, relativePath)
```
Comment (jeromekelleher) [inline ga4gh/datamodel/variants.py]: No - we want to return a HtslibVariantSet here instead of a TabixVariantSet.

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, thanks. You might as well drop the changes on the README, as I'm doing fairly hefty surgery on it at the moment. Also we need to fix the factory as in the comments above.

**Comment by dcolligan** [comment]
@jeromekelleher made those changes

**Comment by jeromekelleher** [comment]
Nice, +1.

**Comment by jeromekelleher** [comment]
Can you rebase this please @dcolligan --- github won't let me merge it.

**Comment by dcolligan** [comment]
@jeromekelleher merge me!

those 1m24s travis builds are sooo much better...

**Comment by macieksmuga** [comment]
Just to test my understanding, this mostly involved replacing the use of `pysam.Tabixfile` with the newer/improved `pysam.VariantFile`? 
As best as I can tell, this is a new addition (as of Feb 9 2015) to pysam, not yet reflected in http://pysam.readthedocs.org/en/latest/api.html#api

**Comment by bioinformed** [comment]
Docs are in the works.  But you do have the author if you have any
concerns, questions or requests.

On Mon, Mar 16, 2015 at 1:54 PM, Maciek Smuga-Otto <notifications@github.com

**Comment by jeromekelleher** [comment]
Basically, that's it @macieksmuga --- the killer feature of the new `pysam.VariantFile` is BCF support, and direct access to htslib's VCF code.

**Pull request merged**